docs/event-processing/idempotent-reader.md

```diff
@@ -0,0 +1,58 @@
+
+## Problem
+How can an application deal with duplicate Events when reading from an Event Stream?
+
+## Solution
+![idempotent-reader](../img/idempotent-reader.svg)
+
+This can be addressed using exactly-once semantics (EOS), including native support for transactions and support for idempotent clients.
+EOS allows [Event Streaming Applications](../event-processing/event-processing-application.md) to process data without loss or duplication, ensuring that computed results are always accurate. 
+
+[Idempotent Writing](idempotent-writer.md) by the [Event Source](../event-source/event-source.md) is the first step in solving this problem. Idempotent Writing provides strong, exactly-once delivery guarantees of the producer's Events, and removes operational failures as a cause of written duplicate Events.
+
+On the reading side, in [Event Processors](../event-processing/event-processor.md) and [Event Sinks](../event-sink/event-sink.md), an Idempotent Reader can be configured to read only committed transactions. This prevents events within incomplete transactions from being read, providing the reader isolation from operational writer failures. Keep in mind that idempotence means that the reader's business logic must be able to process the same consumed event multiple times, where multiple reads have the same effect as a single read of the event. For example, if the reader manages a counter (i.e., a state) for the events it has read, then reading the same event multiple times should only increment the counter once.
+
+Duplicates caused by incorrect application logic from upstream sources are best resolved by fixing the application's logic (i.e., fixing the root cause). In cases where this is not possible, such as when events are generated outside of our control, the next best option is to embed a tracking ID into the event. A tracking ID should be a field that is unique to the logical event, such as an event key or request ID. The consumer can then read the tracking ID, cross-reference it against an internal state store of IDs it has already processed, and discard the event if necessary.
+
+
+## Implementation
+To configure an Idempotent Reader to read only committed transactions, set the following parameter:
+
+```
+isolation.level="read_committed"
+```
+
+In your Kafka Streams application, to handle operational failures, you can [enable EOS](https://docs.confluent.io/platform/current/streams/developer-guide/config-streams.html#processing-guarantee). Within a single transaction, a Kafka Streams application using EOS will atomically update its consumer offsets, its state stores including their changelog topics, its repartition topics, and its output topics.
```

Comment:
Please add how to enable EOS in KStreams:

```
processing.guarantee= exactly_once_v2
```
Unfortunately, I don't know how to create a suggestion for this, as the backticks for the code above interferes with the syntax for the GH suggestion.

Reply:
Btw, I am checking why https://kafka.apache.org/28/documentation/streams/developer-guide/config-streams.html#processing-guarantee says the value should be `exactly_once_beta`. IIRC the `*_v2` name supercedes the `*_beta` name in the latest Kafka, but I have to double-check.

Reply:
Added

Reply:
I don't see the code block with the setting for KStreams? Once added, feel free to merge this PR.

Reply:
@miguno the same config is for both Kafka Streams and ksqlDB, so I reworded and reordered the configuration parameter to make it clearer.